pkg/operator/controllers/subnets/subnetnsg_test.go

```diff
@@ -0,0 +1,361 @@
+)
+
+var (
+	subscriptionId           = "0000000-0000-0000-0000-000000000000"
+	clusterResourceGroupName = "aro-iljrzb5a"
+	infraId                  = "abcd"
+	clusterResourceGroupId   = "/subscriptions/" + subscriptionId + "/resourcegroups/" + clusterResourceGroupName
+	vnetResourceGroup        = "vnet-rg"
+	vnetName                 = "vnet"
+	subnetNameWorker         = "worker"
+	subnetNameMaster         = "master"
+
+	nsgv1NodeResourceId   = clusterResourceGroupId + "/providers/Microsoft.Network/networkSecurityGroups/" + infraId + subnet.NSGNodeSuffixV1
+	nsgv1MasterResourceId = clusterResourceGroupId + "/providers/Microsoft.Network/networkSecurityGroups/" + infraId + subnet.NSGControlPlaneSuffixV1
+	nsgv2ResourceId       = clusterResourceGroupId + "/providers/Microsoft.Network/networkSecurityGroups/" + infraId + subnet.NSGSuffixV2
+)
+
+func getValidClusterInstance() *arov1alpha1.Cluster {
+	return &arov1alpha1.Cluster{
+		Spec: arov1alpha1.ClusterSpec{
+			ArchitectureVersion:    0,
+			ClusterResourceGroupID: clusterResourceGroupId,
+			InfraID:                infraId,
+		},
+	}
+}
+
+func getValidSubnet() *mgmtnetwork.Subnet {
+	return &mgmtnetwork.Subnet{
+		SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
+			NetworkSecurityGroup: &mgmtnetwork.SecurityGroup{
+				ID: to.StringPtr(nsgv1MasterResourceId),
+			},
+		},
+	}
+}
+
+func getMachineObject(name, networkResourceGroup, vnet, subnet string, isMaster bool) (*machinev1beta1.Machine, error) {
+	raw, err := azureproviderv1beta1.EncodeMachineSpec(&azureproviderv1beta1.AzureMachineProviderSpec{
+		NetworkResourceGroup: networkResourceGroup,
+		Vnet:                 vnet,
+		Subnet:               subnet,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	role := "worker"
+	if isMaster {
+		role = "master"
+	}
+
+	return &machinev1beta1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: machineSetsNamespace,
+			Labels: map[string]string{
+				"machine.openshift.io/cluster-api-machine-role": role,
+			},
+		},
+		Spec: machinev1beta1.MachineSpec{
+			ProviderSpec: machinev1beta1.ProviderSpec{
+				Value: raw,
+			},
+		},
+	}, nil
+}
+
+func TestReconcileManager(t *testing.T) {
+	log := logrus.NewEntry(logrus.StandardLogger())
+
+	for _, tt := range []struct {
+		name       string
+		subnetMock func(*mock_subnet.MockManager, *mock_subnet.MockKubeManager)
+		maocli     func() (*maofake.Clientset, error)
+		instance   func(*arov1alpha1.Cluster)
+		wantErr    error
+	}{
+		{
+			name: "Architecture V1 - no change",
+			maocli: func() (*maofake.Clientset, error) {
+				machine1, err := getMachineObject("master", vnetResourceGroup, vnetName, subnetNameMaster, true)
+				if err != nil {
+					return nil, err
+				}
+
+				machine2, err := getMachineObject("worker", vnetResourceGroup, vnetName, subnetNameWorker, false)
+				if err != nil {
+					return nil, err
+				}
+
+				return maofake.NewSimpleClientset(machine1, machine2), nil
```

Comment:
I would just panic in `getMachineObject` when there is an error: it is a developer error so it is ok I think. With panic we can do something like:

```go
maocli: maofake.NewSimpleClientset(
	getMachineObject("master", vnetResourceGroup, vnetName, subnetNameMaster, true),
	getMachineObject("worker", vnetResourceGroup, vnetName, subnetNameWorker, false),
)
```

Reply:
Also, `maocli` function currenlty seems to be exactly the same in all tests cases. Does it make sense to move this into the test body?

Reply:
I dont think it is even needed now when kubeSubnet interface was split...